### DISCUSSION TITLE
In `quarto render`, can I print to stdout (in my terminal, not my notebook) while a loop is running to show my progress? (python)

### DISCUSSION THREAD

**Discussion by bryanwhiting** [quarto-dev/quarto-cli#8501] [Q&A] [answered]
### Description

I have a long process that runs within one cell. I can't break up the process (it's a model training step). I'd like to know what's going on. Can I print out to stdout with `quarto render`? Or would I have to set up a separate logging file and log there?

**Comment by cscheid** [comment, marked as answer]
There's nothing directly available in the way that Quarto communicates with Jupyter that would let you conveniently do that.

Printing to stdout will cause stdout to be collected by Jupyter, placed into a cell, and ultimately into the HTML, so that wouldn't work. I would log to a file and run `tail -F` on that file in a separate terminal.

**Reply by bryanwhiting** [reply]
I've been using a custom file.write() operation on a file to log to a separate file.
